src/telliot_feeds/reporters/tips/listener/funded_feeds_filter.py

```diff
@@ -98,16 +119,27 @@ async def price_change(self, query_id: bytes, value_before: float) -> Optional[f
 
             self.prices[query_id] = value_now[0]
 
-        return _get_price_change(previous_val=value_before, current_val=self.prices[query_id])
+        return _get_price_change(previous_val=value_before_decoded, current_val=self.prices[query_id])
 
     def api_support_check(self, feeds: list[QueryIdandFeedDetails]) -> list[QueryIdandFeedDetails]:
         """Filter funded feeds where threshold is gt zero and no telliot catalog feeds support"""
-        telliot_supported_with_api = [
-            feed
-            for feed in feeds
-            if feed.params.priceThreshold == 0
-            or (feed.query_id in CATALOG_QUERY_IDS and CATALOG_QUERY_IDS[feed.query_id] in CATALOG_FEEDS)
-        ]
+        telliot_supported_with_api = []
+
+        for feed in feeds:
+            if feed.params.priceThreshold == 0:
+                telliot_supported_with_api.append(feed)
+            else:
+                # check first if qtag in CATALOG_FEEDS which means a datafeed exists to check prices
+                datafeed = feed_from_catalog_feeds(feed.query_data)
+                # find query type in DATAFEEDBUILDER Mapping if qtype has generic source ie no manual entry required
+                if datafeed is None:
+                    qtype = decode_typ_name(feed.query_data)
+                    if qtype in TYPES_WITH_GENERIC_SOURCE:
+                        telliot_supported_with_api.append(feed)
+                    else:
+                        continue
+                else:
```

Comment:
This means that if there is not a feed that exists to check prices, it still gets added to `tellilot_supported_with_api`.

You could also refactor a bit like:
```py
def feed_has_api_support(feed):
    # comment why this ensures it's supported..
    if feed.params.priceThreshold == 0:
        return True

    # should exclude feeds that aren't supported in the catalog ?
    datafeed = feed_from_catalog_feeds(feed.query_data)
    if datafeed is None:
        return False

    qtype = decode_typ_name(feed.query_data)
    return qtype in TYPES_WITH_GENERIC_SOURCE


def get_api_supported_feeds(feeds):
    return [feed for feed in feeds if feed_has_api_support(feed)]
```

Reply:
Not sure i understand, but what i'm trying to achieve is for every loop if pt is zero add feed to list. if pt is not zero, check CATALOG_FEEDS,
if CATALOG_FEEDS returns a value then add feed.
But if CATALOG_FEEDS returns None then check if qtype is in the TYPES_WITH_GENERIC_SOURCE list and if in list add feed else continue with loop.

Reply:
Ah, gotcha